store/store.go

```diff
@@ -221,7 +230,9 @@ func (s *store) Set(nodePath string, dir bool, value string, expireTime time.Tim
 		e.PrevNode = prev.Node
 	}
 
-	s.WatcherHub.notify(e)
+	if expireOpts.ExpireTime.IsZero() || !expireOpts.Refresh || ((getErr != nil && getErr.ErrorCode == etcdErr.EcodeKeyNotFound) || n.Value != value) {
```

Comment:
We can adjust this conditional, but if we remove the check for expireOpts.ExpireTime.IsZero(), we'll need to make it an error case when someone refreshes with an expiration of zero otherwise you end up with the potential for a watch never returning even after the key has expired.